### PR TITLE
feat: add syncValueFallback prop for closest-match fallback in syncMethod="value"

### PR DESCRIPTION
**Pull request by roy7**
When using `syncMethod="value"` to synchronize charts with different data arrays (e.g., one chart has 251 trading days and another has 252), the receiving chart hides its tooltip if the synced label doesn't have an exact match in its ticks. This PR adds a new `syncValueFallback` prop that lets users opt into a closest-match fallback instead.

Note that when hovering directly on a sparse chart, the tooltip already shows the nearest data point — this is because direct hover resolves via pixel-coordinate matching (`calculateActiveTickIndex`), which always finds the nearest tick. The sync path with `syncMethod="value"` instead matches by exact string comparison on labels, so without this fallback, synced charts with mismatched data hide the tooltip where direct hover would show one.

## Usage

```tsx
<LineChart syncId="mySync" syncMethod="value" syncValueFallback="closest">
```

## Prop values

- `"none"` (default) — existing behavior, tooltip is hidden when no exact match is found
- `"closest"` — falls back to the nearest tick by string comparison using binary search

## Changes

- New `SyncValueFallback` type (`'none' | 'closest'`) in `src/synchronisation/types.ts`
- New `syncValueFallback` prop on chart components, threaded through Redux state
- `findClosestTick` function: binary search on string-compared tick values
- Fallback is gated by `syncValueFallback === 'closest'` — no behavior change with default props
- Tests covering the new fallback behavior with charts that have overlapping but non-identical data arrays

## Context

This was first discussed in #7020 as a potential follow-up to the tooltip sync counter-emission fix.

## Summary by CodeRabbit

* **New Features**
  * Added a chart-level setting syncValueFallback (default: "none") exposed on Cartesian and Polar charts. When set to "closest", value-based synchronization will pick the nearest tick if an exact label match is missing, improving tooltip sync across charts with slightly mismatched datasets.

* **Tests**
  * Added end-to-end and unit tests covering the closest-tick fallback behavior and its edge cases to ensure reliable synchronized tooltips.

### CODE REVIEW

**Comment by roy7** [comment]
Repro: https://stackblitz.com/edit/vitejs-vite-6negt9nq

Move mouse around in bottom chart, you'll see label always appears. Move it around in the top 2 charts, and you'll see it only appears in bottom chart if you hover over Day 1, 5, or 10.

File: test/synchronisation/useChartSynchronisation.spec.tsx [hunk 1]
```diff
@@ -199,3 +201,59 @@ describe('useTooltipChartSynchronisation', () => {
...
+
+  it('should handle date-like string labels', () => {
+    // ISO date strings sort correctly lexicographically
+    const ticks = [tick('2024-01-01', 0), tick('2024-01-03', 1), tick('2024-01-05', 2)];
+    // '2024-01-02' is between '2024-01-01' and '2024-01-03' — returns previous tick
+    expect(findClosestTick(ticks, '2024-01-02')).toEqual(ticks[0]);
+    // '2024-01-04' is between '2024-01-03' and '2024-01-05' — returns previous tick
+    expect(findClosestTick(ticks, '2024-01-04')).toEqual(ticks[1]);
+  });
+});
```
[the remark this answers is not in the document]
Comment (roy7) [reply test/synchronisation/useChartSynchronisation.spec.tsx:259]: This end-to-end coverage already exists in `test/component/Tooltip/Tooltip.sync.spec.tsx`. The describe block `"when syncMethod=value with syncValueFallback=closest on both charts with slightly different data arrays"` mounts two synced charts with mismatched data and verifies the closest-tick fallback behavior. The counter-emission tests in the same file cover the default `syncValueFallback="none"` case (tooltip hidden when no exact match). (Reply written by Claude Code.)

**Comment by PavelVanecek** [comment]
Instead of adding more and more string enum variants, can we allow the consumer to define? So the `syncMethod` prop will accept a function where the developer can decide however they want to sync and match.

**Comment by roy7** [comment]
Adding support to accept a general function might be a bit beyond what I'm comfortable trying to do, but if you feel the existing behavior is more of a bug than an intended behavior we should keep, I could remove this being an optional variant and instead change the default behavior. My thinking is that if you have synced charts, they should behave the same regardless of which chart you are pointing out. But fixing this would change how it acts for people's existing synced charts if they have charts with different numbers of data points.

**Comment by roy7** [comment]
Closing this one. Looking at it again, `syncMethod` already accepts a function that receives the receiving chart's ticks and the hovered label and returns a tick index, which covers this use case without a new prop. Returning an out-of-range index hides the tooltip, so a "closest within N days, otherwise hide" policy is a few lines in userland, and the right distance measure depends on whether the axis holds dates, numbers or categories, so it belongs there rather than in the library.

The real gap was discoverability, so I've opened #7830 to document the function form and add a "Synchronized Charts With Different Data" example showing exactly this fix.

Thanks for the pointer.
